dnf/lock.py

```diff
@@ -77,8 +77,10 @@ def _try_lock(self):
             os.write(fd, pid)
             os.close(fd)
             return True
-        except OSError:
-            return False
+        except OSError as e:
+            if e.errno == 17:   # File exists
+                return False
+            raise
```

Comment:
"If you need to determine whether an exception was raised but don’t intend to handle it, a simpler form of the raise "statement allows you to re-raise the exception"
(https://docs.python.org/2/tutorial/errors.html#raising-exceptions)
;)